utilities/timeout.jl

```diff
@@ -60,9 +68,86 @@ timer_task = @async begin
 
     # If the process is still running, ask it nicely to terminate
     if isopen(proc)
-        println(stderr, "\n\nProcess failed to exit within $(term_timeout)s, requesting termination and coredump (SIGQUIT) of PID $(proc_pid).")
-        kill(proc, Base.SIGQUIT)
-        println(stderr, "\n\nSent SIGQUIT to PID $(proc_pid).")
+        if use_sigterm
+            println(stderr, "\n\nProcess failed to exit within $(term_timeout)s, requesting termination (SIGTERM) of PID $(proc_pid).")
+            kill(proc, Base.SIGTERM)
+            println(stderr, "\n\nSent SIGTERM to PID $(proc_pid).")
+        else
+            println(stderr, "\n\nProcess failed to exit within $(term_timeout)s, requesting termination and coredump (SIGQUIT) of PID $(proc_pid).")
+            
+            # On Linux, recursively find and signal all descendant processes
```

Comment:
I think a better solution would be to change the launch to put the processes in their own process group, then you can SIGQUIT the entire group at once.